lib/types/command_payload_json.ex

```diff
@@ -0,0 +1,205 @@
+
+  defstruct [:json]
+
+  @impl true
+  def new(%CommandPayload{
+        network_id: network_id,
+        payload: payload,
+        signers: signers,
+        meta: meta,
+        nonce: nonce
+      }) do
+    with {:ok, payload} <- extract_payload(payload),
+         {:ok, meta} <- extract_meta(meta),
+         {:ok, network_id} <- extract_network_id(network_id),
+         {:ok, signers} <- extract_signers_list(signers) do
+      request_body =
+        %{
+          payload: payload,
+          meta: meta,
+          network_id: network_id,
+          nonce: nonce,
+          signers: signers
+        }
+        |> Casex.to_camel_case()
+        |> Jason.encode!()
+
+      %__MODULE__{json: request_body}
+    end
+  end
+
+  @spec extract_network_id(network_id()) :: {:ok, str()}
+  defp extract_network_id(%NetworkID{id: id}), do: {:ok, id}
+
+  @spec extract_payload(pact_payload()) :: {:ok, map()}
+  defp extract_payload(%PactPayload{payload: %ExecPayload{} = exec_payload}) do
+    %ExecPayload{code: %PactCode{code: code}, data: data} = exec_payload
+    payload = %{exec: %{code: code, data: validate_data(data)}}
+    {:ok, payload}
+  end
+
+  defp extract_payload(%PactPayload{
+         payload: %ContPayload{
+           data: data,
+           pact_id: %PactTransactionHash{hash: hash},
+           proof: proof,
+           rollback: %Rollback{value: rollback},
+           step: %Step{number: number}
+         }
+       }) do
+    payload = %{
+      cont: %{
+        data: validate_data(data),
+        pact_id: hash,
+        proof: validate_proof(proof),
+        roollback: rollback,
+        step: number
+      }
+    }
+
+    {:ok, payload}
+  end
+
+  @spec validate_proof(proof()) :: str() | nil
+  defp validate_proof(nil), do: nil
+  defp validate_proof(%Proof{value: proof}), do: proof
+
+  @spec validate_data(data()) :: map() | nil
+  defp validate_data(nil), do: nil
+  defp validate_data(%EnvData{data: data}), do: data
+
+  @spec extract_meta(meta()) :: {:ok, map()}
+  defp extract_meta(%MetaData{
+         creation_time: creation_time,
+         ttl: ttl,
+         gas_limit: gas_limit,
+         gas_price: gas_price,
+         sender: sender,
+         chain_id: %ChainID{id: id}
+       }) do
+    meta = %{
+      chain_id: id,
+      creation_time: creation_time,
+      gas_limit: gas_limit,
+      gas_price: gas_price,
+      sender: sender,
+      ttl: ttl
+    }
+
+    {:ok, meta}
+  end
+
+  @spec extract_signers_list(signers :: signers_list()) :: {:ok, signers()}
+  defp extract_signers_list(%SignersList{signers: list}) do
+    signers = Enum.map(list, fn sig -> extract_signer_info(sig) end)
+    {:ok, signers}
+  end
+
+  @spec extract_signer_info(signer()) :: map()
+  defp extract_signer_info(%Signer{
+         addr: addr,
+         scheme: scheme,
+         pub_key: %Base16String{value: pub_key},
+         clist: %OptionalCapsList{clist: clist}
+       }) do
+    %{
+      addr: validate_addr(addr),
```

Comment:
it is necessary because we need to extract from %Base16String{} the value and the pattern matching will return an error if the addr is nil. so to avoid this error we validate if it is a %Base16String{} we take the value, but if it is nil just return a nil value

Reply:
@santiagoboterokommit that is not a validation :) .... the `%CommandPayload{}` you have as an argument is supposed to be already validated. 

To clarify, replace the "validate" fragment of the function(s) name(s) with a more appropriate one